app/api/tests/test_api.py

```diff
@@ -1199,6 +1201,102 @@ def test_can_filter_doc_without_anotation(self):
+        import_data = []
+        parser = TextUploadAPI.select_parser(import_format)
+        with open(os.path.join(DATA_DIR, filename), 'rb') as f:
+            self.client.post(upload_url, data={'file': f, 'format': import_format})
+            f.seek(0)
+            imported = parser.parse(f)
+            import_data = [elem for elem in [x for elem in imported for x in elem] if elem.get('labels')]
+
+        r = self.client.get(download_url, data={'q': export_format}, HTTP_ACCEPT=response_format)
+        b = BytesIO(r.content)
+        if export_format == 'txt':
+            export_format = import_format
+        parser = TextUploadAPI.select_parser(export_format)
+        exported = parser.parse(b)
+        exported_data = [x for x in [x for elem in exported for x in elem] if x.get('labels') or x.get('annotations')]
+        self.assertTrue(len(import_data) == len(exported_data), 'Length of imported dataset does not match exported')
+        return import_data, exported_data
+
+    # Classification
+    def test_jsonl_classification_import_export_integrity(self):
+        import_data, export_data = self.load_test_helper(upload_url=self.classification_upload_url,
+                                download_url=self.classification_download_url,
+                                filename='classification.jsonl',
+                                import_format='json',
+                                project=self.classification_project,
+                                export_format='json',
+                                response_format='application/json')
+        label_mapping = {label.id: label.text for label in self.classification_project.labels.all()}
+        for im, ex in zip(import_data, export_data):
+            self.assertTrue(im['text'] == ex['text'], 'Integritycheck failed. Items differ.')
+            ex_labels = set(label_mapping[int(x.get('label'))] for x in ex.get('annotations', []))
+            self.assertFalse(set(im.get('labels')).symmetric_difference(ex_labels), 'Integritycheck failed. Items differ.')
+
+    def test_csv_classification_import_export_integrity(self):
+        import_data, export_data = self.load_test_helper(upload_url=self.classification_upload_url,
+                                download_url=self.classification_download_url,
+                                filename='example.csv',
+                                import_format='csv',
+                                project=self.classification_project,
+                                export_format='csv')
+        label_mapping = {label.id: label.text for label in self.classification_project.labels.all()}
+        for im, ex in zip(import_data, export_data):
+            self.assertTrue(im['text'] == ex['text'], 'Integritycheck failed. Items differ.')
+            ex_labels = [label_mapping[int(elem)] for elem in ex.get('labels')]
```

Comment:
define empty list as default for `get()`. otherwise it could fail with 'NoneType' object is not iterable